### PR TITLE
Feature key lineage

### LINKED ISSUE #27: Add zome function that returns all the evolutions of a key
A zome function that returns all the evolutions of a key when given any key from that lineage.

```rs
#[hdk_extern]
pub fn get_key_lineage(
    key_bytes: ByteArray<32>,
) -> ExternResult<Vec<KeyBytes>> {
    // ...
}
```

### LINKED ISSUE #28: Add zome function to check if 2 keys belong to the same lineage
The new function should take 2 keys and return true when both keys are in the same lineage.

```rs
#[hdk_extern]
pub fn same_lineage(
    (key1_bytes, key2_bytes): (ByteArray<32>, ByteArray<32>),
) -> ExternResult<bool> {
    // ...
}
```

### PR DESCRIPTION
**Pull request by mjbrisebois**
Implement zome functions for getting and checking key lineage

resolves #27
resolves #28

### CODE REVIEW

File: zomes/deepkey_csr/src/key_anchor.rs [hunk 1]
```diff
@@ -141,11 +144,113 @@ pub fn get_action_addr_for_key_anchor(
...
+
+
+#[hdk_extern]
+pub fn same_lineage(
+    (key1_bytes, key2_bytes): (ByteArray<32>, ByteArray<32>),
+) -> ExternResult<bool> {
+    let keys = get_key_lineage( key1_bytes )?;
+
+    Ok( keys.contains( &key2_bytes.into_array() ) )
+}
```
Comment (jost-s) [inline zomes/deepkey_csr/src/key_anchor.rs:256]: Why are there 2 identical functions with different names?

**Review by jost-s — APPROVED** [review]
LGTM

File: tests/integration/test_key_management.js [hunk 2]
```diff
@@ -442,6 +487,107 @@ function basic_tests () {
 	}
     });
 
+    it("should query key lineage", async function () {
```
Comment (jost-s) [inline tests/integration/test_key_management.js:490]: 👍

File: zomes/deepkey_csr/src/key_anchor.rs (hunk 1 above)
Comment (mjbrisebois) [reply zomes/deepkey_csr/src/key_anchor.rs:256]: They are not the same.  One calls `query_key_lineage` while the other calls `get_key_lineage`.  The query prefix indicates that it uses source chain querying.  I should write something about that convention in the CONTRIBUTING.md.

File: INTEGRITY_MODEL.md [hunk 3]
```diff
@@ -128,6 +128,8 @@ Validation criteria
 - The record must deserialize into the correct struct
 - Must be a `Update` or `Delete` variant
 - The `original_action_hash` must be the `prior_key_registration` of the revocation
+- There must not be any other `KeyRegistration` on the same source chain referencing the same
+  `prior_key_registration`
 - [Key revocation requirements](#key-revocation)
 - If variant is an `Update`
   - [Key generation requirements](#key-generation)
```
Comment (Kevinduran00) [inline INTEGRITY_MODEL.md:135]: Deep key

**Pull request merged**